openmls/src/framing/mls_content_in.rs

```diff
@@ -0,0 +1,353 @@
+//! This module contains the [`FramedContentIn`] struct and associated helper structs
+//! such as [`FramedContentTbsIn`], as well as their implementations.
+
+use crate::{
+    ciphersuite::signable::Signable,
+    error::LibraryError,
+    group::{GroupEpoch, GroupId},
+    messages::{proposals_in::ProposalIn, CommitIn},
+    versions::ProtocolVersion,
+};
+
+use std::io::{Read, Write};
+
+use super::{
+    mls_auth_content_in::{AuthenticatedContentIn, FramedContentAuthDataIn},
+    mls_content::FramedContentBody,
+    ContentType, Sender, WireFormat,
+};
+
+use serde::{Deserialize, Serialize};
+use tls_codec::{
+    Deserialize as TlsDeserializeTrait, Serialize as TlsSerializeTrait, Size, TlsDeserialize,
+    TlsSerialize, TlsSize, VLBytes,
+};
+
+/// ```c
+/// // draft-ietf-mls-protocol-17
+/// struct {
+///     opaque group_id<V>;
+///     uint64 epoch;
+///     Sender sender;
+///     opaque authenticated_data<V>;
+///
+///     // ... continued in [FramedContentBody] ...
+/// } FramedContent;
+/// ```
+#[derive(
+    Debug, PartialEq, Clone, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
```

Comment:
Same as above. Since this has an -In prefix, I assume it shouldn't implement TlsSerialize.

Reply:
That's needed for `*Tbs` to be serialised. There's no issue with deriving `TlsSerialize`, except that it might be unnecessary in some cases perhaps.